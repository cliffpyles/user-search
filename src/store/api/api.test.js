import { searchUsers, getUserInfo } from './'

test('can search for users', () => {
  return searchUsers({ query: 'ladyleet' }).then(results => {
    const user = results.items.filter(item => item.login === 'ladyleet')

    expect(results.total_count).toBeNumber()
    expect(results.items).toBeArray()
    expect(user.length).toEqual(1)
  })
})

test('can search for users with paginated result', () => {
  return searchUsers({ query: 'Ben', perPage: 5, page: 2 }).then(results => {
    expect(results.items).toBeArray()
    expect(results.items.length).toEqual(5)
    expect(results.links.prev).toBeDefined()
    expect(results.links.next).toBeDefined()
    expect(results.links.last).toBeDefined()
    expect(results.links.first).toBeDefined()
  })
})

test('can get user info', () => {
  return getUserInfo('jayphelps').then(results => {
    expect(results).toMatchSnapshot()
  })
})
