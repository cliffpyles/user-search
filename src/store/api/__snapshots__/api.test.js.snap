// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can get user info 1`] = `
Object {
  "avatar_url": "https://avatars1.githubusercontent.com/u/762949?v=4",
  "bio": null,
  "blog": "https://twitter.com/_jayphelps",
  "company": "prev Netflix ",
  "created_at": "2011-05-02T07:20:04Z",
  "email": null,
  "events_url": "https://api.github.com/users/jayphelps/events{/privacy}",
  "followers": 1306,
  "followers_url": "https://api.github.com/users/jayphelps/followers",
  "following": 35,
  "following_url": "https://api.github.com/users/jayphelps/following{/other_user}",
  "gists_url": "https://api.github.com/users/jayphelps/gists{/gist_id}",
  "gravatar_id": "",
  "hireable": true,
  "html_url": "https://github.com/jayphelps",
  "id": 762949,
  "location": null,
  "login": "jayphelps",
  "name": "Jay Phelps",
  "node_id": "MDQ6VXNlcjc2Mjk0OQ==",
  "organizations_url": "https://api.github.com/users/jayphelps/orgs",
  "public_gists": 24,
  "public_repos": 345,
  "received_events_url": "https://api.github.com/users/jayphelps/received_events",
  "repos_url": "https://api.github.com/users/jayphelps/repos",
  "site_admin": false,
  "starred_url": "https://api.github.com/users/jayphelps/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/jayphelps/subscriptions",
  "type": "User",
  "updated_at": "2019-04-09T19:09:02Z",
  "url": "https://api.github.com/users/jayphelps",
}
`;
